tests/TestServiceTrait.php

```diff
@@ -45,6 +45,7 @@ private function generateTestService($protocol)
         ];
         if ($protocol === 'json') {
             $metadata['jsonVersion'] = "1.1";
+            $metadata['awsQueryCompatible'] = "String";
```

Comment:
I think this is good, I just want to call it out with a note:
This is a widely used file, and now many tests will have this trait added to them.  This makes no difference without adding the header if you look at the logic above, so it's unlikely to effect any other tests as the logic is now.  This sort of opens the door to unintended consequences in test files if the coder isn't aware of what they're doing or just copies and pastes.  This will be fine if we are diligent, especially since the logic around the header being there can't really change.  

Just want to check that @stobrien89 is okay with this as well

Reply:
This is probably fine, but I'd rather see us add `$args = []` as a parameter to `generateTestService` (like generateTestClient) and check for a key that would tell the method to add the property to a json model's metadata.  Then we can just set another service in the provider like 

`$awsQueryCompatibleService =  $this->generateTestService('json', ['queryCompatible' => true])`

Reply:
Sounds good to me. Commit is submitted.